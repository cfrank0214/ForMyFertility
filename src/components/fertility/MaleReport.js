import React, { Component } from 'react';
import { Auth } from 'aws-amplify'
import { Content, Spinner, Text, View } from 'native-base';
import * as Progress from 'react-native-progress';
import MaleFormData from '../../lib/maleformdata';
import Odds from '../../lib/odds';
import { connect } from 'react-redux';

export class MaleReport extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
        };
    }

    async componentDidMount() {
        let user = await Auth.currentAuthenticatedUser();
        
        let maleformdata = new MaleFormData();
        if (this.props.maleQuestions.id) {
            maleformdata = await maleformdata.updateMaleFormData(this.props.maleQuestions);
            this.setState(maleformdata);
        } else {
            maleformdata = await maleformdata.addMaleFormData(this.props.maleQuestions, user.username);
            this.setState(maleformdata);
        }
        this.setState({ given_name: user.attributes.given_name })
        
        // let odds = new Odds();
        // odds = await odds.getOdds(this.state.id, this.props.maleQuestions.currentIVF)
        // this.setState(odds);
    }

    render() {
        return (
            <Content >
                {this.state.maleQuestions &&
                    <View>
                        <Text>{this.state.given_name || ''} your probability of getting a woman pregnant is: </Text>
                        <Progress.Circle
                            progress={0.11}
                            size={300}
                            showsText={true}
                            thickness={10}
                            borderWidth={5}
                            color={'rgb(69, 204, 177)'}
                            filledColor={'rgb(69, 204, 177)'}
                        />
                    </View>}
                
                {!this.state.maleQuestions &&
                    <View>
                        <Spinner />
                        <Text>Getting Report!</Text>

                    </View>}
            </Content>
        );
    }
}

export default connect(({ maleQuestions }) => ({ maleQuestions }))(MaleReport);
