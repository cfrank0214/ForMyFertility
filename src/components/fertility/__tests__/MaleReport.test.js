/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MaleReport } from '../MaleReport';
import { View, Spinner, Text } from 'native-base';

describe('>>>MaleReport --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper
  
  beforeEach(()=>{
      wrapper = shallow(<MaleReport />)
      
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <MaleReport />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('+++ render the DUMB component', () => {
     expect(wrapper.length).toEqual(1)
  });
  it('contains 1 <View /> component', () => {
    expect(wrapper.find(View).length).toBe(1);
  });
  it('contains 1 <Spinner /> component', () => {
    expect(wrapper.find(Spinner).length).toBe(1);
  }); 
  it('contains 1 <Text /> component', () => {
    expect(wrapper.find(Text).length).toBe(1);
  }); 
});