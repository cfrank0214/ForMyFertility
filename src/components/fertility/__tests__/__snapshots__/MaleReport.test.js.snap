// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>MaleReport --- Shallow Render REACT COMPONENTS renders correctly 1`] = `null`;
